myhpi/core/models.py

```diff
@@ -29,6 +30,11 @@ class BasePage(Page):
         FieldPanel("is_public", widget=forms.CheckboxInput),
         FieldPanel("visible_for", widget=forms.CheckboxSelectMultiple),
     ]
+    # FilterFields required for restricting search results
+    search_fields = Page.search_fields + [
+        index.FilterField("group_id"),
```

Comment:
do we really want to search for the group id?

Reply:
No we don't. This field is required for permission checks (so we don't output search results the user can't visit)

Reply:
I don't understand why this is working, as searching on filtered QuerySets should not be possible as of https://github.com/wagtail/wagtail/issues/6616. As ist works right now, I would be fine with accepting that even if it generates warnings